test/Templates.Test/BaselineTest.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Testing.xunit;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Templates.Test
+{
+    public class BaselineTest : TemplateTestBase
+    {
+        public BaselineTest(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        public static TheoryData<string, string[]> TemplateBaselines
+        {
+            get
+            {
+                using (var stream = typeof(BaselineTest).Assembly.GetManifestResourceStream("Templates.Test.template-baselines.json"))
+                {
+                    using (var jsonReader = new JsonTextReader(new StreamReader(stream)))
+                    {
+                        var baseline = JObject.Load(jsonReader);
+                        var data = new TheoryData<string, string[]>();
+                        foreach (var template in baseline)
+                        {
+                            foreach (var authOption in (JObject)template.Value)
+                            {
+                                data.Add(
+                                    (string)authOption.Value["Arguments"],
+                                    ((JArray)authOption.Value["Files"]).Select(s => (string)s).ToArray());
+                            }
+                        }
+
+                        return data;
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TemplateBaselines))]
+        public void Template_Produces_The_Right_Set_Of_Files(string arguments, string[] expectedFiles)
```

Comment:
template-baselines.json contains files for mvc, razor, webapi and their different authentication options along with the list of files that they produce.

This test runs through each of the combinations and verifies that the right set of files is present.